crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -174,6 +175,64 @@ impl ConfigSettingAccessor<C8yUrlSetting> for TEdgeConfig {
     }
 }
 
+impl ConfigSettingAccessor<C8yHttpSetting> for TEdgeConfig {
+    #[allow(deprecated)]
+    fn query(&self, _setting: C8yHttpSetting) -> ConfigSettingResult<HostPort<443>> {
+        self.data
+            .c8y
+            .http
+            .as_ref()
+            .cloned()
+            .or(self.data.c8y.url.as_ref().cloned().map(ConnectUrl::into))
+            .ok_or(ConfigSettingError::ConfigNotSet {
+                key: C8yUrlSetting::KEY,
```

Comment:
Wouldn't be better to tell that `C8yHttpSetting` is not set rather than `C8yUrlSetting`?

```suggestion
                key: C8yHttpSetting::KEY,
```

Reply:
My rationale is that because `C8yHttpSetting` and `C8yMqttSetting` inherit their values from `c8y.url`, I think of them more as optional overrides rather than settings in their own right. When both `c8y.http` and `c8y.mqtt` are set, then indeed `c8y.url` is not necessary. However, if this error is received by the user, the suggestion to set `c8y.http` is helpful only if `c8y.mqtt` is also already set, which if it isn't, the user will receive a second error.

So in short, IMHO `c8y.url` is sort of "required", and in the error message it makes more sense to tell the user to set a required setting, instead of an optional one.

Reply:
It makes sense. Ideally, it would be better to tell the user to set either one or the other. But this is not necessary for now.